lte/gateway/python/magma/pipelined/rpc_servicer.py

```diff
@@ -573,12 +589,29 @@ def GetAllTableAssignments(self, request, context):
         Get the flow table assignment for all apps ordered by main table number
         and name
         """
+        self._log_grpc_payload(request)
         table_assignments = self._service_manager.get_all_table_assignments()
         return AllTableAssignments(table_assignments=[
             TableAssignment(app_name=app_name, main_table=tables.main_table,
                             scratch_tables=tables.scratch_tables) for
             app_name, tables in table_assignments.items()])
 
+    # --------------------------
+    # Internal
+    # --------------------------
+
+    def _log_grpc_payload(self, grpc_request):
+        if not self._print_grpc_payload:
+            return
+        if not grpc_request:
+            return
+        indent = '  '
+        dbl_indent = indent + indent
```

Comment:
using https://docs.python.org/3/library/pprint.html might be better.

Reply:
Also it will be great if you can store last 100 grpc request/payload in some sort of the list. With the https://github.com/magma/magma/pull/3313 we should be able to dump this information even if logging is not enabled.

Reply:
I've used pprint/pformat initially but it doesn't get the same indent as this. Added a queue with 100 grpc reqs, lmk what you think